pkg/services/rpcsrv/server_test.go

```diff
@@ -74,22 +74,22 @@ type rpcTestCase struct {
 }
 
 const genesisBlockHash = "0f8fb4e17d2ab9f3097af75ca7fd16064160fb8043db94909e00dd4e257b9dc4"
-const testContractHash = "565cff9508ebc75aadd7fe59f38dac610ab6093c"
-const deploymentTxHash = "a14390941cc3a1d87393eff720a722e9cd350bd6ed233c5fe2001326c80eb68e"
+const testContractHash = "449fe8fbd4523072f5e3a4dfa17a494c119d4c08"
+const deploymentTxHash = "bbb8ec059dd320dc9de5a8fb8c75351d8e369fca0256f7a6bdb623dcf71861ed"
 
 const (
 	verifyContractHash                = "06ed5314c2e4cb103029a60b86d46afa2fb8f67c"
 	verifyContractAVM                 = "VwIAQS1RCDBwDBTunqIsJ+NL0BSPxBCOCPdOj1BIskrZMCQE2zBxaBPOStkoJATbKGlK2SgkBNsol0A="
-	verifyWithArgsContractHash        = "4dc916254efd2947c93b11207e8ffc0bb56161c5"
-	nnsContractHash                   = "892429fcd47c30f8451781acc627e8b20e0d64f3"
+	verifyWithArgsContractHash        = "6261b3bf753bdc3d24c1327a23fd891e1c8a7ccd"
+	nnsContractHash                   = "450d1918a72fef97b48096bfec8d749961deef55"
 	nnsToken1ID                       = "6e656f2e636f6d"
-	nfsoContractHash                  = "730ebe719ab8e3b69d11dafc95cdb9bf409db179"
+	nfsoContractHash                  = "6b8d711017b81dc975a23ddd7a7320fac5e9ee06"
 	nfsoToken1ID                      = "7e244ffd6aa85fb1579d2ed22e9b761ab62e3486"
-	storageContractHash               = "ebc0c16a76c808cd4dde6bcc063f09e45e331ec7"
+	storageContractHash               = "2e0809e43dc0f81dab1b48e45dccbc1148ee2a2e"
```

Comment:
@AnnaShaleva could you advise, please, I think tests are failing because `storageContractHash` is incorrect or maybe was deployed wrongly. However `TestCreateBasicChain` hadn't any errors.

Reply:
You need to regenerate basic chain with `saveChain = true`, then fetch relevant data from the test logs (including storage contract hash) and update RC server test variables. From what I see, the following test is failing:
```
        --- FAIL: TestRPC/websocket/batch_with_single_request (0.38s)
            --- FAIL: TestRPC/websocket/batch_with_single_request/invokefunction (0.01s)
                --- FAIL: TestRPC/websocket/batch_with_single_request/invokefunction/positive,_with_storage_changes (0.00s)
                    server_test.go:1434: 
                        	Error Trace:	/home/anna/Documents/GitProjects/nspcc-dev/neo-go/pkg/services/rpcsrv/server_test.go:1434
                        	            				/home/anna/Documents/GitProjects/nspcc-dev/neo-go/pkg/services/rpcsrv/server_test.go:2541
                        	Error:      	elements differ
                        	            	
                        	            	extra elements in list A:
                        	            	([]interface {}) (len=2) {
                        	            	 (dboper.Operation) {
                        	            	  State: (string) (len=7) "Changed",
                        	            	  Key: ([]uint8) (len=5) {
                        	            	   00000000  fa ff ff ff 0b                                    |.....|
                        	            	  },
                        	            	  Value: ([]uint8) (len=7) {
                        	            	   00000000  54 b2 d2 a3 51 79 12                              |T...Qy.|
                        	            	  }
                        	            	 },
                        	            	 (dboper.Operation) {
                        	            	  State: (string) (len=7) "Changed",
```

It happens because its results are non-deterministic if you regenerate test chain. To fix it just update expected storage values for this test.
Regarding the second test:
```
        logger.go:146: 2024-11-08T10:29:22.479+0300	INFO	Error encountered with rpc request	{"code": -108, "method": "traverseiterator", "params": "[f4f4c870-a739-4856-83de-e36795314d16 16406a2a-8065-494a-8892-9b572a9f77f5 1]"}
        --- FAIL: TestRPC/http/traverseiterator (0.02s)
            --- FAIL: TestRPC/http/traverseiterator/good (0.00s)
                server_test.go:3071: 
                    	Error Trace:	/home/anna/Documents/GitProjects/nspcc-dev/neo-go/pkg/services/rpcsrv/server_test.go:3071
                    	Error:      	Not equal: 
                    	            	expected: 99
                    	            	actual  : 2
                    	Test:       	TestRPC/http/traverseiterator/good
```
We haven't changed storage contract in any way, and `storageContractHash` contains correct hash because it's the only contract with `iterateOverValues` method in the whole repo. Try to regenerate chain one more time and check which values are really stored in the contract storage. Also, try to regenerate chain on master and see if the same error is reproduced.

Reply:
On master, the error is reproduced (changed saveChain to true, generated .acc, changed back saveChain to false,  run tests). TestClient_IteratorSessions, TestClient_Iterator_SessionConfigVariations - failed. 
The same as I have within the newly generated one on nep24.

Reply:
> On master the error is reproduced

Create a separate bug for this, and let's fix it firstly. I won't say that the problem is with contract hash, it's something iterator-related (although we didn't touch this code for quite a long time).

Reply:
Ref. #3671.